ui/components/Filters.js

```diff
@@ -0,0 +1,429 @@
+  }, []);
+
+  /**
+   * fetchPatterns constructs the queries based on the parameters given
+   * and fetches the patterns
+   * @param {number} page current page
+   * @param {number} pageSize items per page
+   * @param {string} search search string
+   * @param {string} sortOrder order of sort
+   */
+  function fetchPatterns(page, pageSize, search, sortOrder) {
+    if (!search) search = "";
+    if (!sortOrder) sortOrder = "";
+
+    const query = `?page=${page}&page_size=${pageSize}&search=${encodeURIComponent(search)}&order=${encodeURIComponent(
+      sortOrder
+    )}`;
+
+    updateProgress({ showProgress: true });
+
+    dataFetch(
+      `/api/experimental/pattern${query}`,
+      {
+        credentials: "include",
+      },
+      (result) => {
+        console.log("PatternFile API", `/api/experimental/pattern${query}`);
+        updateProgress({ showProgress: false });
+        if (result) {
+          setPatterns(result.patterns || []);
+          setPage(result.page || 0);
+          setPageSize(result.page_size || 0);
+          setCount(result.total_count || 0);
+        }
+      },
+      handleError
+    );
+  }
+
+  function handleError(error) {
+    updateProgress({ showProgress: false });
+
+    enqueueSnackbar(`There was an error fetching results: ${error}`, {
+      variant: "error",
+      action: function Action(key) {
+        return (
+          <IconButton key="close" aria-label="Close" color="inherit" onClick={() => closeSnackbar(key)}>
+            <CloseIcon />
+          </IconButton>
+        );
+      },
+      autoHideDuration: 8000,
+    });
+  }
+
+  function resetSelectedRowData() {
+    return () => {
+      setSelectedRowData(null);
+    };
+  }
+
+  function handleSubmit(data, id, name, type) {
+    if (type === "delete") {
+      dataFetch(
+        `/api/experimental/pattern/${id}`,
+        {
+          credentials: "include",
+          method: "DELETE",
+        },
+        () => {
+          console.log("PatternFile API", `/api/experimental/pattern/${id}`);
+          updateProgress({ showProgress: false });
+          fetchPatterns(page, pageSize, search, sortOrder);
+        },
+        handleError
+      );
+    }
+
+    if (type === "update") {
+      dataFetch(
+        `/api/experimental/pattern`,
+        {
+          credentials: "include",
+          method: "POST",
+          body: JSON.stringify({ pattern_data: { id, pattern_file: data }, save: true }),
+        },
+        () => {
+          console.log("PatternFile API", `/api/experimental/pattern`);
+          updateProgress({ showProgress: false });
+          fetchPatterns(page, pageSize, search, sortOrder);
+        },
+        handleError
+      );
+    }
+
+    if (type === "upload") {
+      dataFetch(
+        `/api/experimental/pattern`,
+        {
+          credentials: "include",
+          method: "POST",
+          body: JSON.stringify({ pattern_data: { pattern_file: data }, save: true }),
```

Comment:
I believe the request body needs to be a different for filters POST API for now // @navendu

Reply:
We have not added a POST endpoint yet. And yes, it will be different